tests/converters/app-converter.test.ts

```diff
@@ -0,0 +1,308 @@
+import { AppConverter, StackConverter } from '../../src/converters/app-converter';
+import { Stack } from 'aws-cdk-lib/core';
+import { StackComponentResource, StackOptions } from '../../src/types';
+import * as path from 'path';
+import * as mockfs from 'mock-fs';
+import * as pulumi from '@pulumi/pulumi';
+import { MockCallArgs, MockResourceArgs } from '@pulumi/pulumi/runtime';
+import { Bucket, BucketPolicy } from '@pulumi/aws-native/s3';
+import { createStackManifest } from '../utils';
+
+// Convert a pulumi.Output to a promise of the same type.
+function promiseOf<T>(output: pulumi.Output<T>): Promise<T> {
+    return new Promise((resolve) => output.apply(resolve));
+}
+
+function setMocks() {
+    pulumi.runtime.setMocks(
+        {
+            call: (args: MockCallArgs): { [id: string]: any } => {
+                switch (args.token) {
+                    case 'aws-native:index:getAccountId':
+                        return {
+                            accountId: '12345678910',
+                        };
+                    case 'aws-native:index:getRegion':
+                        return {
+                            region: 'us-east-2',
+                        };
+                    case 'aws-native:index:getPartition':
+                        return {
+                            partition: 'aws',
+                        };
+                    case 'aws-native:index:getAzs':
+                        return {
+                            azs: ['us-east-1a', 'us-east-1b'],
+                        };
+                    default:
+                        return {};
+                }
+            },
+            newResource: (args: MockResourceArgs): { id: string; state: any } => {
+                return {
+                    id: args.name + '_id',
+                    state: {
+                        ...args.inputs,
+                        arn: args.name + '_arn',
+                    },
+                };
+            },
+        },
+        'project',
+        'stack',
+        false,
+    );
+}
+
+class MockStackComponent extends StackComponentResource {
+    public readonly name = 'stack';
+    public readonly assemblyDir: string;
+    public stack: Stack;
+    public options?: StackOptions | undefined;
+    constructor(dir: string) {
+        super('stack');
+        this.assemblyDir = dir;
+        this.registerOutputs();
+    }
+
+    registerOutput(outputId: string, output: any): void {}
+}
+
+describe('App Converter', () => {
```

Comment:
Reading this out loud, if I understand this right, this builds out some manifests that CDK would build, and feeds them into AppConverter to assert that the right Pulumi resources are created against the mocks. There is an implied CDK program that would build these manifests. WDYT about having that CDK program kept in the test suite and the manifests recomputed? Perhaps wouldn't work or is too high level?

Reply:
We could do that, but I prefer having the lower level tests rather than super high level tests. We had some very high level tests already and it made it very hard to know what areas were under test and which were not. I also think it helps with understanding what is being converted, otherwise you don't have anything to reference.